plugins/builtin/apm/nomad/plugin/node.go

```diff
@@ -0,0 +1,215 @@
+	}
+	a.logger.Debug("performing node pool APM query", "query", q)
+
+	// Identify the resource available and consumed within the target pool.
+	resources, err := a.getPoolResources(query.poolIdentifier)
+	if err != nil {
+		return 0, err
+	}
+	a.logger.Debug("collected node pool resource data",
+		"allocated_cpu", resources.allocated.cpu, "allocated_memory", resources.allocated.mem,
+		"allocatable_cpu", resources.allocatable.cpu, "allocatable_memory", resources.allocatable.mem)
+
+	var result float64
+
+	// There is no need for a default catch all here as the metric has been
+	// validated during the query parsing.
+	switch query.metric {
+	case queryMetricMem:
+		if resources.allocatable.mem == 0 {
+			return 0, errors.New("zero allocatable memory found in pool")
+		}
+		result = calculateNodePoolResult(float64(resources.allocated.mem), float64(resources.allocatable.mem))
+	case queryMetricCPU:
+		if resources.allocatable.cpu == 0 {
+			return 0, errors.New("zero allocatable cpu found in pool")
+		}
+		result = calculateNodePoolResult(float64(resources.allocated.cpu), float64(resources.allocatable.cpu))
+	}
+
+	return result, nil
+}
+
+// getPoolResources gathers the allocatable and allocated resources for the
+// specified node pool. Any error in calling the Nomad API for details will
+// result in an error. This is because with missing data, we cannot reliably
+// make calculations.
+func (a *APMPlugin) getPoolResources(id *scaleutils.PoolIdentifier) (*nodePoolResources, error) {
+
+	nodes, _, err := a.client.Nodes().List(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list Nomad nodes: %v", err)
+	}
+
+	// Perform our node filtering so we are left with a list of nodes that form
+	// our pool and that are in the correct state.
+	nodePoolList, err := id.IdentifyNodes(nodes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to identify nodes within pool")
+	}
+	if len(nodePoolList) == 0 {
+		return nil, errors.New("no nodes identified within pool")
+	}
+
+	// Ensure we instantiate the whole object.
+	resp := nodePoolResources{
+		allocatable: &poolResources{},
+		allocated:   &poolResources{},
+	}
+
+	// Loop here so we only need to iterate the list of nodes once.
+	for _, node := range nodePoolList {
+
+		// If we get a single error when performing the following lookups we
+		// cannot reliably make calculations.
+		if err := a.getNodeAllocatableResources(node.ID, resp.allocatable); err != nil {
+			return nil, fmt.Errorf("failed to get allocatable resources on node %s: %v", node.ID, err)
+		}
+		if err := a.getNodeAllocatedResources(node.ID, resp.allocated); err != nil {
+			return nil, fmt.Errorf("failed to get allocated resources on node %s: %v", node.ID, err)
+		}
+	}
+
+	return &resp, nil
+}
+
+// getNodeAllocatableResources updates the poolResources tracking with the
+// allocatable resources on the node.
+func (a *APMPlugin) getNodeAllocatableResources(nodeID string, pool *poolResources) error {
```

Comment:
no big deal but... why "allocatable" and not "free"?

Reply:
It isn't the free resources on the node, but the total amount of resources that can be allocated.

Reply:
Since the can of paint is out 😁 ....

`allocatable` and `allocated` are too similar. I had to do some double take in some lines, so it might be a source of weird typo bugs. 

Is `usable` a good alternative? Or maybe `consumable`?

Reply:
Yeah, naming here is tricky here. We can keep it as it is then, makes sense to use the same term as what Nomad telemetry exports.